src/context/AppProvider/AppContext.tsx

```diff
@@ -101,6 +101,14 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
     require(`dayjs/locale/${selectedLocale}.js`)
   }, [selectedLocale])
 
+  useEffect(() => {
+    if (!wallet) return
+    const walletSupportedChains = Object.values(KnownChainIds).filter(chainId =>
+      walletSupportsChain({ chainId, wallet, isSnapInstalled }),
+    )
+    dispatch(portfolio.actions.setWalletSupportedChainIds(walletSupportedChains))
+  }, [dispatch, isSnapInstalled, wallet])
```

Comment:
Is moving this to `<AppContext />` done to be able to have selectors? Worried about the rerender implications of this, since `wallet` tends to referentially change quite a lot because of keystore

Reply:
Nice, handled in f7f92b85de39b7b046c071d1ac247dc09c79ec74 by checking equivalence by value in the action before mutating state